src/blr/cli/bamtofq.py

```diff
@@ -0,0 +1,96 @@
+"Writes fq files from bam files"
+
+import logging
+import pysam
+
+from tqdm import tqdm
+import dnaio
+
+from blr import utils
+
+logger = logging.getLogger(__name__)
+
+
+def main(args):
+    logger.info(f"Starting analysis")
+    query_cache = dict()
+    summary = Summary()
+    out_interleaved = not args.r2_out
+    with pysam.AlignmentFile(args.bam, "rb") as reader, \
+            dnaio.open(args.r1_out, file2=args.r2_out, interleaved=out_interleaved, mode="w",
+                       fileformat="fastq") as writer:
+        for query in tqdm(reader):
+            summary.tot_reads_in += 1
+
+            # Read pair cache system
+            if query.query_name not in query_cache:
+                query_cache[query.query_name] = query
+                continue
+            else:
+                query_1 = query_cache[query.query_name]
+                query_2 = query
+                del query_cache[query.query_name]
```

Comment:
You can remove this line and instead write.

```
query_1 = query_cache.pop(query.query_name)
```

Reply:
Also there is a looooot of different names starting with query in this section. I find it a little confusing. Possibly rename:
query_cache --> pair_cache
query_1 --> mate
query_2 --> query (i.e don't rename).

Reply:
Can't say I have a strong opinion so I'll abide.